lib/source-map.js

```diff
@@ -362,22 +361,54 @@ SourceMap.prototype._scanMappings = function(srcMap, sourcesOffset, namesOffset,
   this.content.mappings = mappings;
 };
 
-SourceMap.prototype.end = function() {
-  if (this.mapCommentType === 'line') {
-    this.stream.write('//# sourceMappingURL=' + this.mapURL);
-  } else {
-    this.stream.write('/*# sourceMappingURL=' + this.mapURL + ' */');
-  }
-  mkdirp.sync(path.dirname(this.mapFile));
-  fs.writeFileSync(this.mapFile, JSON.stringify(this.content));
+SourceMap.prototype.end = function(cb, thisArg) {
+  var stream = this.stream;
+  var sourceMap = this;
+
+
   return new RSVP.Promise(function(resolve, reject) {
-    this.stream.on('finish', resolve);
-    this.stream.on('error', reject);
-    this.stream.end();
-  }.bind(this));
+    stream.on('error', function(error) {
+      stream.on('close', function() {
+        reject(error);
+      });
+
+      stream.end();
+    });
+
+    var error, success;
+
+    try {
+      if (cb) {
+        cb.call(thisArg, sourceMap);
```

Comment:
no, that callback is what populates `this.content` that is written out in `writeFileSync`